File: backend/app/schemas/ai.py
from pydantic import BaseModel


class GenerateRequest(BaseModel):
    job_description: str | None = None
    target_role: str | None = None


class ImproveSummaryRequest(BaseModel):
    summary: str


class RewriteSummaryRequest(BaseModel):
    summary: str


class ExperienceInput(BaseModel):
    company: str
    position: str
    description: str
    start_date: str = ""
    end_date: str = ""


class GenerateExperienceRequest(BaseModel):
    experience: ExperienceInput


class ImproveExperienceRequest(BaseModel):
    experience: ExperienceInput


class ProjectInput(BaseModel):
    title: str
    description: str
    start_date: str = ""
    end_date: str = ""


class ImproveProjectRequest(BaseModel):
    project: ProjectInput


class SuggestSkillsRequest(BaseModel):
    skills: list[str]


class GenerateResponse(BaseModel):
    resume_data: dict
    cover_letter: dict
    ats_analysis: dict


class SummaryResponse(BaseModel):
    summary: str


class ExperienceResponse(BaseModel):
    experience: ExperienceInput


class ProjectResponse(BaseModel):
    project: ProjectInput


class SkillsResponse(BaseModel):
    skills: list[str]

File: frontend/types/ai.ts
export interface ATSAnalysis {
  id?: string;
  resume_id?: string;
  overall_score: number;
  strengths: string[];
  weaknesses: string[];
  recommendations: string[];
  missing_keywords: string[];
  analyzed_at?: string;
}

export interface GenerateResponse {
  resume_data: Record<string, unknown>;
  cover_letter: Record<string, unknown>;
  ats_analysis: ATSAnalysis;
}

export interface CoverLetter {
  id: string;
  company_name: string;
  job_title: string;
  content: string;
  created_at?: string;
}

export interface CoverLetterListResponse {
  items: CoverLetter[];
}

export interface CoverLetterResponse {
  id: string;
  company_name: string;
  job_title: string;
  content: string;
  created_at?: string;
}

export interface SummaryResponse {
  summary: string;
}

export interface ExperienceInput {
  company: string;
  position: string;
  description: string;
}

export interface ExperienceResponse {
  experience: ExperienceInput;
}

export interface ProjectInput {
  title: string;
  description: string;
}

export interface ProjectResponse {
  project: ProjectInput;
}

export interface SkillsResponse {
  skills: string[];
}

export interface FillFieldsResponse {
  resume_data: Record<string, unknown>;
}
